apps/src/aichat/redux/aichatRedux.ts

```diff
@@ -457,6 +459,7 @@ export const submitChatContents = createAsyncThunk(
     }
 
     thunkAPI.dispatch(clearChatMessagePending());
+    dispatch(sendInProgressReport('aichat'));
```

Comment:
Would it make more sense to move these dispatches closer to the specific user interaction and/or state change that triggers them? I'm just wary of these being buried in these long thunk functions and us missing a future update case. For example, we could instead call this on the `handleSubmit` callback in the user chat message editor, so this gets fired whenever the user clicks the submit button?

For the save case, I think where you have it makes sense, assuming we want to only report progress after a successful save. If instead we want to report progress as soon as the user attempts a save, then we could have the dispatch trigger at the beginning of the `saveAiCustomization` thunk function?

Reply:
> For the save case, I think where you have it makes sense, assuming we want to only report progress after a successful save. If instead we want to report progress as soon as the user attempts a save, then we could have the dispatch trigger at the beginning of the saveAiCustomization thunk function?

Yeah - I was thinking that in progress meant a change to the model customizations or a successful submit so I placed the report after a successful save. @samantha-code @dancodedotorg - any opinions?

Reply:
> Would it make more sense to move these dispatches closer to the specific user interaction and/or state change that triggers them? I'm just wary of these being buried in these long thunk functions and us missing a future update case. For example, we could instead call this on the handleSubmit callback in the user chat message editor, so this gets fired whenever the user clicks the submit button?

Sure, that makes sense. I wasn't sure about the best placement of this report so moving it so that it's easier for future updates sounds good!

Reply:
> Yeah - I was thinking that in progress meant a change to the model customizations or a successful submit so I placed the report after a successful save. @samantha-code @dancodedotorg - any opinions?

This makes sense to me - feels okay to not save progress if there's an error of some kind.

Reply:
Thanks @dancodedotorg!
I was considering this for the chat message submission as well  so this was a reason I placed it after assistant response was received. I could add comments to highlight this. Lmk what you think @sanchitmalhotra126.

Reply:
Yep if we only want to report an attempt after a response was successfully received, then here makes sense. Let's add a comment for clarity